### ISSUE TITLE
Testing code in docs does not compile

### ISSUE THREAD

**Issue by austinrivas** [seanmonstar/warp#611]
I'm evaluating warp for a faas project and was just looking to get a simple filter with a testing suite running.

Just copy / pasta'ing the code in the [warp::test docs](https://docs.rs/warp/0.2.3/warp/test/index.html) throws a compilation error on `cargo test` using `cargo 1.44.0-nightly (8751eb301 2020-04-21)`

```rust
use warp::Filter;

fn sum() -> impl Filter<Extract = (u32,), Error = warp::Rejection> + Copy  {
    warp::path::param()
        .and(warp::path::param())
        .map(|x: u32, y: u32| {
            x + y
        })
}

fn math() -> impl Filter<Extract = (String,), Error = warp::Rejection> + Copy {
    warp::post()
        .and(sum())
        .map(|z: u32| {
            format!("Sum = {}", z)
        })
}

#[cfg(test)]
mod tests {
  use super::*;

  #[test]
  fn test_sum() {
    let filter = sum();

    // Execute `sum` and get the `Extract` back.
    let value = warp::test::request()
        .path("/1/2")
        .filter(&filter)
        .unwrap();
    assert_eq!(value, 3);

    // Or simply test if a request matches (doesn't reject).
    assert!(
        !warp::test::request()
            .path("/1/-5")
            .matches(&filter)
    );
  }
}
```

```sh
error[E0599]: no method named `unwrap` found for opaque type `impl core::future::future::Future` in the current scope
  --> src/lib.rs:27:12
   |
27 |           .unwrap();
   |            ^^^^^^ method not found in `impl core::future::future::Future`

error[E0600]: cannot apply unary operator `!` to type `impl core::future::future::Future`
  --> src/lib.rs:32:11
   |
32 | /           !warp::test::request()
33 | |               .path("/1/-5")
34 | |               .matches(&filter)
   | |_______________________________^ cannot apply unary operator `!`

error: aborting due to 2 previous errors
```

**Comment by jxs** [comment]
Hi, yeah should be:
```Rust
use warp::Filter;

fn sum() -> impl Filter<Extract = (u32,), Error = warp::Rejection> + Copy  {
    warp::path::param()
        .and(warp::path::param())
        .map(|x: u32, y: u32| {
            x + y
        })
}

fn math() -> impl Filter<Extract = (String,), Error = warp::Rejection> + Copy {
    warp::post()
        .and(sum())
        .map(|z: u32| {
            format!("Sum = {}", z)
        })
}

#[cfg(test)]
mod tests {
  use super::*;

  #[tokio::test]
  async fn test_sum() {
    let filter = sum();

    // Execute `sum` and get the `Extract` back.
    let value = warp::test::request()
        .path("/1/2")
        .filter(&filter)
        .await
        .unwrap();
    assert_eq!(value, 3);

    // Or simply test if a request matches (doesn't reject).
    assert!(
        !warp::test::request()
            .path("/1/-5")
            .matches(&filter)
    );
  }
}
```
want to submit a PR fixing it?

**Comment by austinrivas** [comment]
no problem, I'll submit one soon

**Comment by calvinbrown085** [comment]
@austinrivas @jxs Can we close this?

**Comment by austinrivas** [comment]
Yeah go ahead, thanks guys

**Comment by jxs** [comment]
fixed on #718
